packages/test-sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/UserTests.kt

```diff
@@ -111,6 +111,34 @@ class UserTests {
         }
     }
 
+    @Test
+    fun getProviderType() = runBlocking {
+        val email = randomEmail()
+        val emailUser = createUserAndLogin(email, "123456")
+        assertEquals(AuthenticationProvider.EMAIL_PASSWORD, emailUser.provider)
+    }
+
```

Comment:
Are these tests sufficient?

Reply:
Generally, we try to test:

1) Success case
2) Boundaries
3) Error cases

2/3 doesn't really apply in this case though, except maybe the cases I mentioned where the user is either logged out or deleted.

So, yes, I would say this is probably sufficient.